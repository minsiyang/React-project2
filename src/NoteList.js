import React, { Component } from 'react';

class NoteList extends Component {
  constructor(props) {
    super(props);
    this.state = { input: '', notes: [] };

    this.handleChange = this.handleChange.bind(this);

    this.handleSubmit = this.handleSubmit.bind(this);
  }

  handleChange(event) {
    this.setState( { input: event.target.value } )
  }

  handleSubmit(event) {
    this.state.notes.push(this.state.input)
    this.setState({notes: this.state.notes});
    this.setState({input: ''})
    event.preventDefault();
  }

  render(){
    return (
      <div className="NoteList">
        <form>
          <label>
            Add Note: <input type="text" value={this.state.input} onChange={this.handleChange} />
          </label>
          <input type="submit" value="Submit" onClick={this.handleSubmit} />
        </form>
        <ul>{this.state.notes.map((note, index) => {
          return <li key={index}>{note}</li>
        })}</ul>
      </div>
    )
  }
}

export default NoteList;