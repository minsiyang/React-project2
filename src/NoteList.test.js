import React from 'react';
import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import renderer from 'react-test-renderer';

import NoteList from './NoteList';


test('user can add a note', () => {
  render(<NoteList />);
  
  const textBox = screen.getByRole('textbox');
  userEvent.type(textBox, "A test note!")
  userEvent.click(screen.getByText('Submit'))
  const note = screen.getByText('A test note!');

  expect(note).toBeInTheDocument();

})

test('renders correctly when there are no notes', () => {
  const tree = renderer.create(<NoteList />).toJSON();
  expect(tree).toMatchSnapshot();
});