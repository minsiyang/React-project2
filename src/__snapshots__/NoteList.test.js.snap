// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when there are no notes 1`] = `
<div
  className="NoteList"
>
  <form>
    <label>
      Add Note: 
      <input
        onChange={[Function]}
        type="text"
        value=""
      />
    </label>
    <input
      onClick={[Function]}
      type="submit"
      value="Submit"
    />
  </form>
  <ul />
</div>
`;
